app/src/main/java/ch/hikemate/app/utils/MapUtils.kt

```diff
@@ -2,108 +2,134 @@ package ch.hikemate.app.utils
+   * https://wiki.openstreetmap.org/wiki/Zoom_levels
+   *
+   * @param bounds The bounding box (minLat, maxLat, minLon, maxLon) defining the area for which to
+   *   calculate the zoom level.
+   * @return An integer representing the best zoom level (lower value means zoomed out, higher value
+   *   means zoomed in).
+   */
+  fun calculateBestZoomLevel(bounds: Bounds): Int {
+    val minLong = bounds.minLon
+    val maxLong = bounds.maxLon
+    val minLat = bounds.minLat
+    val maxLat = bounds.maxLat
 
-  return GeoPoint(centerLat, centerLong)
-}
+    val centerLat = (minLat + maxLat) / 2
+
+    val latDiff = maxLat - minLat
+    val longDiff = maxLong - minLong
+
+    // Adjust longitude difference based on the latitude compression factor
+    val adjustedLongDiff = longDiff * cos(Math.toRadians(centerLat))
+
+    // Calculate the maximum degree difference between lat and adjusted long
+    val maxDegreeDiff = kotlin.math.max(latDiff, adjustedLongDiff)
 
-/**
- * Calculates the best zoom level for a given bounding box. The zoom level is determined based on
- * the largest degree difference between the latitude and longitude within the bounds, considering
- * latitude compression due to the Earth's spherical shape. This function selects a zoom level that
- * fits the provided area, ensuring the best visibility for the given bounds in a map view.
- *
- * The levels in the var zoomLevels are taken from OSMs 'Zoom levels' documentation:
- * https://wiki.openstreetmap.org/wiki/Zoom_levels
- *
- * @param bounds The bounding box (minLat, maxLat, minLon, maxLon) defining the area for which to
- *   calculate the zoom level.
- * @return An integer representing the best zoom level (lower value means zoomed out, higher value
- *   means zoomed in).
- */
-fun calculateBestZoomLevel(bounds: Bounds): Int {
-  val minLong = bounds.minLon
-  val maxLong = bounds.maxLon
-  val minLat = bounds.minLat
-  val maxLat = bounds.maxLat
-
-  val centerLat = (minLat + maxLat) / 2
-
-  val latDiff = maxLat - minLat
-  val longDiff = maxLong - minLong
-
-  // Adjust longitude difference based on the latitude compression factor
-  val adjustedLongDiff = longDiff * cos(Math.toRadians(centerLat))
-
-  // Calculate the maximum degree difference between lat and adjusted long
-  val maxDegreeDiff = kotlin.math.max(latDiff, adjustedLongDiff)
-
-  val zoomLevels =
-      listOf(
-          360.0,
-          180.0,
-          90.0,
-          45.0,
-          22.5,
-          11.25,
-          5.625,
-          2.813,
-          1.406,
-          0.703,
-          0.352,
-          0.176,
-          0.088,
-          0.044,
-          0.022,
-          0.011,
-          0.005,
-          0.003,
-          0.001,
-          0.0005,
-          0.00025)
-
-  val bestZoomLevel =
-      zoomLevels.indexOfFirst { it <= maxDegreeDiff }.takeIf { it != -1 } ?: (zoomLevels.size - 1)
-
-  return bestZoomLevel
+    val zoomLevels =
```

Comment:
Nitpick: rename this list to something such as `degreeCoverageIndexedByZoom`, since it is not a list of zoom levels per se.

Reply:
Done